modules/core/lib/Controller/Login.php

```diff
@@ -43,11 +64,391 @@ public function __construct(
+     * username/password authentication.
+     *
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @return \SimpleSAML\XHTML\Template
+     */
+    public function loginuserpass(Request $request): Template
+    {
+        // Retrieve the authentication state
+        if (!$request->query->has('AuthState')) {
+            throw new Error\BadRequest('Missing AuthState parameter.');
+        }
+        $authStateId = $request->query->get('AuthState');
+
+        /** @var array $state */
+        $state = $this->authState::loadState($authStateId, UserPassBase::STAGEID);
+
+        /** @var \SimpleSAML\Module\core\Auth\UserPassBase|null $source */
+        $source = $this->authSource::getById($state[UserPassBase::AUTHID]);
+        if ($source === null) {
+            throw new Exception(
+                'Could not find authentication source with id ' . $state[UserPassBase::AUTHID]
+            );
+        }
+
+        if ($request->request->has('username')) {
+            $username = $request->request->get('username');
+        } elseif (
+            $source->getRememberUsernameEnabled()
+            && $request->cookies->has($source->getAuthId() . '-username')
+        ) {
+            $username = $request->cookies->get($source->getAuthId() . '-username');
+        } elseif (isset($state['core:username'])) {
+            $username = (string) $state['core:username'];
+        } else {
+            $username = '';
+        }
+
+        if ($request->request->has('password')) {
+            $password = $request->request->get('password');
+        } else {
+            $password = '';
+        }
+
+        $errorCode = null;
+        $errorParams = null;
+        $queryParams = [];
+
+        if (isset($state['error'])) {
+            $errorCode = $state['error']['code'];
+            $errorParams = $state['error']['params'];
+            $queryParams = ['AuthState' => $authStateId];
+        }
+
+        $cookie = null;
+        if (!empty($request->request->get('username')) || !empty($password)) {
+            // Either username or password set - attempt to log in
+
+            if (array_key_exists('forcedUsername', $state)) {
+                $username = $state['forcedUsername'];
+            }
+
+            if ($source->getRememberUsernameEnabled()) {
+                if (
+                    $request->request->has('remember_username')
+                    && ($request->request->get('remember_username') === 'Yes')
+                ) {
+                    $expire = time() + 31536000;
+                } else {
+                    $expire = time() - 300;
+                }
+
+                $cookie = new Cookie(
+                    $source->getAuthId() . '-username',
+                    $username, // value
+                    $expire, // expire
+                    '/', // path
+                    '', // domain
+                    true, // secure
+                    true, // httponly
+                );
```

Comment:
Should these things not derive from config options we have for cookies?
Maybe these could be handled centrally in the Cookie class instead of at every point that sets a cookie?

Reply:
We have settings for the session-cookie and the language-cookie.. Nothing else..
We _could_ use `Utils\HTTP::isHttps()` to figure out if we need to set the `secure` flag to true or false, but is there really a valid reason today to set it to false? httponly is a flag that depends on the specific cookie....

Reply:
Alright, maybe not in scope now then. Maybe we could at some point make these cookie settings more generic instead of config per-cookie.